doc/nrf/ug_bootloader.rst

```diff
@@ -92,7 +92,96 @@ These signed variants can be used to perform an upgrade procedure through the :r
+The |NCS| includes a sample implementation of an immutable bootloader.
+As upgradable bootloader, it provides and extends :doc:`mcuboot:index`.
 
-See :doc:`mcuboot:index` for information on how to implement an upgradable bootlader.
+Both bootloaders can easily be included in your application using :ref:`ug_multi_image`.
+
+Adding an immutable bootloader
+==============================
+
+To add an immutable bootloader to your application, set :option:`CONFIG_SECURE_BOOT` and add your private key file under :option:`CONFIG_SB_SIGNING_PUBLIC_KEY`.
+|how_to_configure|
+
+See the documentation of the :ref:`bootloader` sample for more information.
+The :ref:`bootloader_build_and_run` section has detailed instructions for adding the immutable bootloader as first stage of the secure bootloader chain.
+
+Adding an upgradable bootloader
+===============================
+
+To add an upgradable bootloader to your applications, set :option:`CONFIG_BOOTLOADER_MCUBOOT`.
+|how_to_configure|
+
+The upgradable bootloader should always be used in combination with the immutable bootloader.
+
+.. note::
+   It is possible to include this bootloader without the immutable bootloader.
+   However, in this case the upgradable bootloader cannot actually be upgraded, because the root of trust is missing.
+
+   In the default configuration, most |NCS| samples include the upgradable bootloader but not the immutable bootloader.
+   The reason for this is that the immutable bootloader must be configured with a private key before use.
+
+The |NCS| uses MCUboot as upgradable bootloader.
+See :doc:`mcuboot:index` for information about the default implementation.
 :ref:`mcuboot:mcuboot_ncs` gives details on the integration of MCUboot in |NCS|.
+
+You can configure MCUboot by setting configuration options for the ``mcuboot`` child image.
+
+Flash partitions used by MCUboot
+--------------------------------
+
+MCUboot requires two image slots: one that contains the application to be booted (the *primary slot*), and one where a new application can be stored before it is activated (the *secondary slot*).
+See the *Image Slots* section in the :doc:`MCUboot documentation <mcuboot:design>` for more information.
+
+The |NCS| variant of MCUboot uses the :ref:`partition_manager` to configure the flash partitions for these image slots.
+
+In the default configuration, the Partition Manager dynamically sets up the partitions as required.
+If you want to control where in memory the flash partitions are placed, you can define static partitions for your application.
+See :ref:`ug_pm_static` for more information.
+
+It is possible to use external flash as the storage partition for the secondary slot.
+This requires a driver for the external flash that supports:
+
+* Single-byte read and write
+* Writing data to flash
```

Comment:
```suggestion
* Writing data stored in internal flash

Reply:
I don't know how to properly formulate this requirement. But it requires that the driver is able to write data directly from flash. So usually it is done by internally copying the data to RAM in the driver if it uses DMA.

Reply:
You mean copying data from internal flash to external flash?

Reply:
Yes. So most drivers that use DMA usually requires you to pass in a ram buffer but sometimes MCUBoot uses data which are stored as constants in internal flash.